test/e2e/storage/persistent_volumes-vsphere.go

```diff
@@ -142,16 +143,46 @@ var _ = SIGDescribe("PersistentVolumes:vsphere", func() {
 
 		Test Steps:
 		1. Delete PVC.
-		2. Delete POD, POD deletion should succeed.
+		2. Verify Volume remains attached to the pod after deleting claim.
+		3. Verify Volume is accessible in the pod after deleting claim.
+		4. Verify associated PV is present and its status should be failed.
+		5. Delete Pod and wait until PV is unmounted and detached from the Node.
+		6. Wait and Verify PV is deleted after POD is deleted.
 	*/
 
 	It("should test that deleting a PVC before the pod does not cause pod deletion to fail on vsphere volume detach", func() {
 		By("Deleting the Claim")
 		framework.ExpectNoError(framework.DeletePersistentVolumeClaim(c, pvc.Name, ns), "Failed to delete PVC ", pvc.Name)
 		pvc = nil
 
+		By("Verify the Volume is attached to the Node")
+		// Get the copy of the Pod to know the assigned node name.
+		pod, err := c.CoreV1().Pods(ns).Get(clientPod.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		isVolumeAttached, verifyDiskAttachedError := verifyVSphereDiskAttached(vsp, pv.Spec.VsphereVolume.VolumePath, types.NodeName(pod.Spec.NodeName))
+		Expect(verifyDiskAttachedError).NotTo(HaveOccurred())
+		Expect(isVolumeAttached).To(BeTrue())
+
+		pvs := make([]*v1.PersistentVolume, 1)
+		pvs[0] = pv
+		By("Verify the volume is accessible and available in the pod")
+		verifyVSphereVolumesAccessible(pod, pvs, vsp)
+		framework.Logf("Verified that Volume is accessible in the POD after deleting PV claim")
+
+		// Verify PV is Present, after PVC is deleted and PV status should be Failed.
+		pv, err := c.CoreV1().PersistentVolumes().Get(pvc.Spec.VolumeName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(pv.Status.Phase == v1.VolumeFailed).To(BeTrue())
```

Comment:
Don't we have to retry this? Because there is no guarantee that by the time this executed pv controller has marked PV as failed.

Reply:
```
Expect(pv.Status.Phase).To(Equal(v1.VolumeFailed))
```

Reply:
added retry
```
Expect(framework.WaitForPersistentVolumePhase(v1.VolumeFailed, c, pv.Name, 1*time.Second, 60*time.Second)).NotTo(HaveOccurred())
```